packages/firestore/test/unit/local/test_remote_document_cache.ts

```diff
@@ -26,32 +34,49 @@ import { DocumentKey } from '../../../src/model/document_key';
  * A wrapper around a RemoteDocumentCache that automatically creates a
  * transaction around every operation to reduce test boilerplate.
  */
-export class TestRemoteDocumentCache {
-  constructor(
-    public persistence: Persistence,
-    public cache: RemoteDocumentCache
+export abstract class TestRemoteDocumentCache {
+  protected constructor(
+    private readonly persistence: Persistence,
+    protected cache: RemoteDocumentCache
   ) {}
 
+  /**
+   * Reads all of the documents first so we can safely add them and keep the size calculation in
+   * sync.
+   */
   addEntries(maybeDocuments: MaybeDocument[]): Promise<void> {
     return this.persistence.runTransaction(
       'addEntry',
       'readwrite-primary',
       txn => {
-        return this.cache.addEntries(txn, maybeDocuments);
+        const changeBuffer = this.cache.newChangeBuffer();
+        return PersistencePromise.forEach(maybeDocuments, maybeDocument =>
+          changeBuffer.getEntry(txn, maybeDocument.key).next()
```

Comment:
It really seems undesirable for consumers to have to call getEntry() before calling addEntry().  I'm a little surprised our code already does this in all cases?

Reply:
That's what makes byte counting reasonably cheap to do; we were already doing this in all cases (except some tests).